src/System.CommandLine/Command.cs

```diff
@@ -74,6 +74,8 @@ private protected override void AddSymbol(Symbol symbol)
 
         public ICommandHandler Handler { get; set; }
 
+        public virtual Type HandlerType { get; }
```

Comment:
Could this HandlerType be removed with usage like ICommandHandler::Handler.GetType() instead?

Reply:
Not if `Handler` is null.

Reply:
Right, so on line 95 just do a handler is null check and call gettype when handler is not null. I don't see any other usage of handler type outside the context of this scope:
https://github.com/dotnet/command-line-api/blob/631b88ee984944467cd590fa43cf438e5e3be27a/src/System.CommandLine.Hosting/HostingExtensions.cs#L95